readthedocs/rtd_tests/tests/test_sync_versions.py

```diff
@@ -152,6 +152,362 @@ def test_new_tag_update_inactive(self):
         version_8 = Version.objects.get(slug='0.8.3')
         self.assertFalse(version_8.active)
 
+    def test_normal_behavior_for_stable_after_deleting_user_defined_tag(self):
+        """
+        The user creates a tag named ``stable`` on an existing repo,
+        when syncing the versions, the RTD's ``stable`` is lost
+        and doesn't update automatically anymore, when the tag is deleted
+        on the user repository, the RTD's ``stable`` is back.
+        """
+        version8 = Version.objects.create(
+            project=self.pip,
+            identifier='0.8.3',
+            verbose_name='0.8.3',
+            type=TAG,
+            active=False,
+            machine=False,
+        )
+        self.pip.update_stable_version()
+        current_stable = self.pip.get_stable_version()
+
+        # 0.8.3 is the current stable
+        self.assertTrue(
+            version8.identifier,
+            current_stable.identifier
+        )
+        self.assertTrue(current_stable.machine)
+
+        version_post_data = {
+            'branches': [
+                {
+                    'identifier': 'origin/master',
+                    'verbose_name': 'master',
+                },
+            ],
+            'tags': [
+                # User new stable
+                {
+                    'identifier': '1abc2def3',
+                    'verbose_name': 'stable',
+                },
+                {
+                    'identifier': '0.8.3',
+                    'verbose_name': '0.8.3',
+                },
+            ],
+        }
+
+        resp = self.client.post(
+            '/api/v2/project/{}/sync_versions/'.format(self.pip.pk),
+            data=json.dumps(version_post_data),
+            content_type='application/json',
+        )
+        self.assertEqual(resp.status_code, 200)
+
+        self.pip.refresh_from_db()
```

Comment:
I'm not sure if this is necessary here.

Reply:
As a general rule:

> `refresh_from_db` is needed when you have a db object already in memory and the registry in the db is changed by an external reason (celery, api request, HTTP post request, etc). So, in that case, the object in memory has to be refreshed from the db to have the new values.

Reply:
Related fields (those one that generate new queries to db) are not affected by this. For example, `self.pip.versions.all()` will hit the db and it's not affected by an old `self.pip` in memory.